### PR TITLE
Update /stations API to provide information on variables associated with a station

### PR DESCRIPTION
**Pull request by corviday**
API expansion to allow querying on a station and variable and getting the first and last timestamps for available observations.

Work needed:

- [x] fit format to what the front end expects
- [x] possibly use a materialized view instead of a query to get the timestamps (query hasn't been slow so far, but haven't done thorough testing)
- [x] consider whether there might be a more RESTful approach - hard to make REST fit the station/history/variable model. Look at variable per history instead of per station?
- [x] tests
- [x] documentation

### CODE REVIEW

**Comment by corviday** [comment]
Question for @Nospamas :

Do you think `station_id` should be included in the output of `/station/{stationid}/variables/{variableid}`? It makes sense to me to do that, since the information returned is about the variable at that station (ie, includes start and end date for observations at the station).

**Comment by corviday** [comment]
Notes from discussions with @Nospamas elsewhere:

* we agree that` /station/{station_id}/variables/{variable_id}` should return a `station_id`
* some stations, for example `1938`, are currently very slow; using a materialized view for observation minimum and maximum timestamps should be investigated

File: sdpb/api/stations.py [hunk 1]
```diff
@@ -287,7 +299,90 @@ def collection(
...
+####
+
+
+def get_station_variables(station_id=None):
+    assert station_id is not None
+    session = get_app_session()
+    q = session.query(Station).select_from(Station).filter(Station.id == station_id)
+    q = add_station_network_publish_filter(q)
+    station = q.one()
+    station_histories_etc = (
```
Comment (rod-glover) [inline sdpb/api/stations.py:354]: This exact query is repeated in the function `single`. Suggest DRYing both up by extracting to a function.

File: sdpb/api/stations.py [hunk 2]
```diff
@@ -287,7 +299,90 @@ def collection(
 
 
 ####
-# /stations/{station_id}/observations/{var_id}
+# /stations/{station_id}/variables/{var_id}
+####
```
Comment (rod-glover) [inline sdpb/api/stations.py:303]: Would it be helpful to put the station-variables api into a separate module, say `station_variables.py`? It's okay here but it makes for a very long module.

File: sdpb/api/stations.py [hunk 3]
```diff
@@ -26,9 +26,11 @@
 from flask import url_for
 from sqlalchemy import func
 from pycds import Station, History, StationObservationStats, Obs, Variable
+from pycds.orm.native_matviews.version_3505750d3416 import VarsPerHistory
```
Comment (rod-glover) [inline sdpb/api/stations.py:29]: As discussed

**Review by rod-glover — APPROVED** [review]
This looks sensible and is evidently working.

**Comment by Nospamas** [comment]
I've pulled from #82 into this branch which fixes the postgres version on live, looks like there may still be some tests to fix, but I think once fixed it should fix the workflow errors.

**Pull request merged**